Modules/_tkinter.c

```diff
@@ -360,21 +360,35 @@ Sleep(int milli)
 }
 #endif /* MS_WINDOWS */
 
-/* Wait up to 1s for the mainloop to come up. */
+/* Wait for the mainloop to come up. */
+static int mainloopwaitattempts = 10; // 1 second
 
 static int
 WaitForMainloop(TkappObject* self)
 {
-    int i;
-    for (i = 0; i < 10; i++) {
-        if (self->dispatching)
-            return 1;
+    int i = 0;
+    int max_attempts = mainloopwaitattempts;
+    if (self->dispatching)
+    {
+        return 1;
+    }
+    for (i = 0; i < max_attempts; i++) {
         Py_BEGIN_ALLOW_THREADS
         Sleep(100);
         Py_END_ALLOW_THREADS
+        if (self->dispatching) 
+        {
+            return 1;
+        }
     }
-    if (self->dispatching)
-        return 1;
+    PyErr_WarnEx(PyExc_DeprecationWarning,
+        "It seems you are waiting for Tcl events "
+        "to be dispatched.\n"
+        "Future behavior will wait indefinitely. Adjust the wait time "
+        "limit with setmainloopwaitattempts().\n"
+        "If you want to avoid this wait consider polling "
+        "dispatching() before issuing commands from a thread.\n", 1
+    );
```

Comment:
I think it makes more sense to be here considering my recent change to the planned future behavior.  Hitting this error after one second is the actual deprecated behavior, rather than waiting at all.